carbon-design-system/carbon-addons-iot-react
# fix(table): view management

Closes https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2409, https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2410, https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2411

**Summary**

- When user view management is enabled and the `onUserViewModified` callback is fired for search value changes, the *true* search value is not given back.  Fixed the useEffect to handle this

- TableViewDropdown's `View all` option is basically a 'default' view and should never be able to get overwritten. I've removed the option to be able to 'Save changes' and only provided the 'Save as new view' instead. The power of overwriting default views should be given to the consumer as they would most likely make a separate "default" view that could be overwritten, but still preserve the *true* default meaning, show the original table without any config

- TableViewDropdown's actions should be at the top of the list so that users don't need to scroll the entire list (if there are many views) to get to the actions

**Change List (commits, features, bugs, etc)**

- Fixed the existing TableViewDropdown story to use the correct edited prop
- Added additional TableViewDropdown stories to show more use cases
- Organized TableViewDropdown's tests into better defined categories
- Updated and added new tests for TableViewDropdown to reflect the above changes

**Acceptance Test (how to verify the PR)**

- Go to http://127.0.0.1:3000/?path=/story/1-watson-iot-table--stateful-example-with-create-save-views , add a breakpoint in the `onUserViewModified` and begin to type into the search bar. You should see the values being logged and the rendered values as the same value. (previously this would stop showing the true values)
- Go to http://127.0.0.1:3000/?path=/story/1-watson-iot-%E2%9A%A0%EF%B8%8F-table-tableviewdropdown--table-view-dropdown-edited and ensure that there is no 'Save changes' button
- Go to http://127.0.0.1:3000/?path=/story/1-watson-iot-%E2%9A%A0%EF%B8%8F-table-tableviewdropdown--table-view-dropdown-edited and ensure that the actions are now at the top of the list

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Ensure that you can still see the 'Save changes' button when using custom views in edited mode with this story: http://127.0.0.1:3000/?path=/story/1-watson-iot-%E2%9A%A0%EF%B8%8F-table-tableviewdropdown--custom-view-selected
- Check the default search value stories. I'm still not really sure what the purpose of that prop was
